### ISSUE TITLE
Refactor API key config organization

### ISSUE THREAD

**Issue by csssuf** [tjcsl/cslbot#1332]
API key configuration is kind of a mess. I think there's a better way to organize them, but I'm not sure what that is. We should figure that out and implement it.

**Comment by csssuf** [comment]
Most of my motivation behind this is for API keys only needed by specific commands. Those definitely don't need to be in the core config, and we should look into creating more generalized per-command configuration apart from API keys.

**Comment by pefoley2** [comment]
I mean, I feel like having a central place for bot-specific configuration is a good thing.
I guess we could add something like groups.cfg, but I'm not really sure how that would improve anything over the status quo.
I feel like it's easier to add keys to a central file, rather than having to go dig through a config directory or something.
That said, I'm open to suggestions.

**Comment by csssuf** [comment]
Yeah, thinking about it more I think it makes more sense to keep everything in one config file but reorganize how it's laid out and how it gets passed to commands.

**Comment by pefoley2** [comment]
Sure, the current system more or less grew organically and could do w/ some cleanup.
